rs/nervous_system/candid_utils/src/printing/tests.rs

```diff
@@ -0,0 +1,78 @@
+use super::*;
+
+#[derive(candid::CandidType, Clone, PartialEq)]
+struct DummyCandidStruct {
+    pub status: Option<i32>,
+    pub module_hash: Vec<u8>,
+    pub controllers: String,
+    pub memory_size: Option<u64>,
+    pub cycles: Option<u64>,
+}
+
+#[derive(candid::CandidType, Clone, PartialEq)]
+enum DummyCandidVariant {
+    Foo(String),
+    Bar { abc: String, xyz: String },
+}
+
+#[derive(candid::CandidType, Clone, PartialEq)]
+struct DummyCandidContainer {
+    foo: DummyCandidVariant,
+    bar: Result<DummyCandidVariant, String>,
+}
+
+#[track_caller]
+fn run_test<T: CandidType>(value: T, expected_result: Result<String, String>) {
+    let observed_result = pretty(value);
+
+    match (observed_result, expected_result) {
+        (Ok(observed), Ok(expected)) => {
+            pretty_assertions::assert_str_eq!(observed, expected);
```

Comment:
Perhaps surprisingly, you can do

```
use pretty_assertions::assert_str_eq;
```

at the top of this file. I think this is more conventional, for better or worse.